ksql-version-metrics-client/src/main/java/io/confluent/ksql/version/metrics/KsqlVersionCheckerAgent.java

```diff
@@ -59,7 +61,8 @@ public void start(final KsqlModuleType moduleType, final Properties ksqlProperti
                   ksqlVersionCheckerConfig,
                   serverRuntime,
                   moduleType,
-                  enableSettlingTime
+                  enableSettlingTime,
+                  activeChecker
```

Comment:
If you switch `KsqlVersionChecker` over to just accepting a `Supplier<Boolean>` for the activityCheck, then you can add a method for calculating if we're active or not:

```
private boolean isActive() {
    if (hasActiveQueries()) {
        return true;
    }

    final long lastRequestAge = clock.get() - lastRequestTimestamp;
    return lastInterval < MAX_INTERVAL;
}
```

And replace this line with:

```suggestion
                  this::isActive
```